feat: add video ids and transcripts for working with regex lectures

scissorsneedfoodtoo opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!-- Feel free to add any additional description of changes below this line -->
This PR adds in video ids wherever necessary, and either adds in a transcript or converts a current description into a transcript for all the lectures in this block.

One major change is the title for one lecture, from "What Are Lookaheads...?" to "What Are Lookahead and Lookbehind Assertions, and How Do They Work?" I believe "lookaheads" was a typo in that case.

I also tried to replace all the images with code blocks. Many of the images show console logs, and I tried to represent those with comments. Sometimes I had to take a few liberties to show RegExp objects similar to the way they appear in Chrome's console / screenshots in the video.

scissorsneedfoodtoo committed:
fix: add video ids for lectures in the working with regex block

scissorsneedfoodtoo committed:
chore: update title and slug for the lookahead and lookbehind lecture

scissorsneedfoodtoo committed:
feat: add transcripts and modify existing ones, replace images with code blocks

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5dc74176e4c496d09e6.md:
```diff
@@ -1,14 +1,18 @@
-# videoId: nVAaxZ34khk
-dashedName: what-are-lookaheads-and-lookbehind-assertions-and-how-do-they-work
+videoId: U-XnBty3JOo
+dashedName: what-are-lookahead-and-lookbehind-assertions-and-how-do-they-work
```
it looks like we are not changing dashed names

https://github.com/freeCodeCamp/freeCodeCamp/issues/58314#issuecomment-2613380651

scissorsneedfoodtoo replied:
Awesome, thanks for confirming that. I'll go in and revert these changes now.

jdwilkin4 reviewed (commented):
Looks good.
Left one comment 👍

scissorsneedfoodtoo committed:
fix: revert change to lookahead and lookbehind assertions lecture slug

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5d0048bb74c18431296.md:
```diff
@@ -2,31 +2,35 @@
 id: 6733c5d0048bb74c18431296
 title: What Are Character Classes, and What Are Some Common Examples?
 challengeType: 19
```
```suggestion
challengeType: 11
```

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5dc74176e4c496d09e6.md:
```diff
@@ -1,14 +1,18 @@
 id: 6733c5dc74176e4c496d09e6
-title: What Are Lookaheads and Lookbehind Assertions, and How Do They Work?
+title: What Are Lookahead and Lookbehind Assertions, and How Do They Work?
 challengeType: 19
```
```suggestion
challengeType: 11
```

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5e54e3a154c8078ed48.md:
```diff
@@ -2,17 +2,21 @@
 id: 6733c5e54e3a154c8078ed48
 title: What Are Regex Quantifiers, and How Do They Work?
 challengeType: 19
```
```suggestion
challengeType: 11
```

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5f20cc9584cada942a4.md:
```diff
@@ -2,49 +2,61 @@
 id: 6733c5f20cc9584cada942a4
 title: What Are Capturing Groups and Backreferences, and How Do They Work?
 challengeType: 19
```
```suggestion
challengeType: 11
```

Dario-DC reviewed (commented):
I just realized some video ids have been added without changing the challenge type. I checked that only.

scissorsneedfoodtoo committed:
Update curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5d0048bb74c18431296.md

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

scissorsneedfoodtoo committed:
Update curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5dc74176e4c496d09e6.md

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

scissorsneedfoodtoo committed:
Update curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5e54e3a154c8078ed48.md

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

scissorsneedfoodtoo committed:
Update curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5f20cc9584cada942a4.md

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

scissorsneedfoodtoo commented:
Awesome, thanks for catching that, @Dario-DC. I didn't realize that we needed to change these challenge types when adding in the video ids.

I'll double check my last PR where I updated the transcripts and added in the video ids to make sure those have the correct `challengeType`, too.

Dario-DC commented:
@scissorsneedfoodtoo  Some have been added without changing the challenge type in previous PRs but I should have updated all of them in #58590

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5dc74176e4c496d09e6.md:
````diff
@@ -62,11 +87,25 @@ const regex = /(?<!free)code/i;
+); // ['code', index: 33, input: 'I need someone for free to write code for me', groups: undefined]
+```
 
 Notice how even though our regular expression uses a lookbehind to check for the presence of free, it does not match free. The only text included in the match is code.
````
```suggestion
Notice how even though our regular expression uses a lookbehind to check for the presence of `free`, it does not match `free`. The only text included in the match is `code`.
```

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5e54e3a154c8078ed48.md:
````diff
@@ -86,15 +119,31 @@ Unfortunately, we've just realized we read the requirements wrong. We need to al
 ```
 
-But our pattern is getting long again. Thankfully, there's another short-hand for "match the previous character zero or more times" –  the asterisk symbol. Let's replace our quantifier with that in the pattern, and test it:
+But our pattern is getting long again. Thankfully, there's another short-hand for "match the previous character zero or more times" –  the asterisk (`*`) symbol. Let's replace our quantifier with that in the pattern, and test it:
````
```suggestion
But our pattern is getting long again. Thankfully, there's another short-hand for "match the previous character zero or more times" –  the asterisk symbol (`*`). Let's replace our quantifier with that in the pattern, and test it:
```
More a question than a suggestion. Is it normal in English to place the actual symbol there? To me it sounds more natural to have it after the word `symbol`.

jdwilkin4 replied:
I have seen it both ways. So it should be fine to leave it as is

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5e54e3a154c8078ed48.md:
```diff
@@ -86,15 +119,31 @@ Unfortunately, we've just realized we read the requirements wrong. We need to al
 Now we successfully match any identifier with zero or more letters followed by four to six numbers. But it turns out this is crashing our system – we actually have to require at least one letter.
 
-Again, we could use a quantifier with a minimum of one and no defined maximum, or we could use yet another special syntax – the plus symbol:
+Again, we could use a quantifier with a minimum of one and no defined maximum, or we could use yet another special syntax – the plus (`+`) symbol:
```
```suggestion
Again, we could use a quantifier with a minimum of one and no defined maximum, or we could use yet another special syntax – the plus symbol (`+`):
```
Same as before

jdwilkin4 replied:
I have seen it both ways. So it should be fine to leave it as is

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5f20cc9584cada942a4.md:
````diff
@@ -1,50 +1,62 @@
 ```
 
-This works on its own, but what if we didn't know how many `o`'s were in "`code`"? If we need a quantifier for one or more `o`'s…
+This works on its own, but what if we didn't know how many `o`'s were in `code`? If we need a quantifier for one or more `o`'s:
````
```suggestion
This works on its own, but what if we didn't know how many `o`'s were in `code`? If we need a quantifier for one or more `o`s:
```

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5f20cc9584cada942a4.md:
```diff
@@ -53,13 +65,16 @@ console.log("freecoooooooodecamp".replace(regex, "paidcodeworld"));
-This is where a backreference comes in. Instead of hardcoding the "`code`" portion of our replacement string, we can reference the captured group directly.
+This is where a backreference comes in. Instead of hardcoding the `code` portion of our replacement string, we can reference the captured group directly.
 
 In a replace call, you achieve a backreference by using a dollar sign (`$`) followed by the number of the capture group to use.
 
 In our case, that would be `$1`, since code is captured in the first capture group.
```
```suggestion
In a `replace` call, you achieve a backreference by using a dollar sign (`$`) followed by the number of the capture group to use. In our case, that would be `$1`, since `code` is captured in the first capture group:
```

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5f20cc9584cada942a4.md:
````diff
@@ -53,13 +65,16 @@ console.log("freecoooooooodecamp".replace(regex, "paidcodeworld"));
+```
 
 We have now successfully preserved an unknown number of `o` characters when converting `freecodecamp` into `paidcodeworld`. But backreferences aren't just limited to the replace call. You can actually use them directly in a regular expression.
 
````
"`o`'s" instead of "`o`s" in the sentence below.

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5f20cc9584cada942a4.md:
````diff
@@ -77,7 +92,11 @@ This current expression won't ensure that the number of `o` characters is the sa
+console.log(regex.test("freecooooodecamp is great i love freecodecamp")); // false
+```
 
 And with that, we can see that a string with the correct number of `o`'s matches, while a string with two different numbers of `o`'s does not.
````
```suggestion
And with that, we can see that a string with the correct number of `o`s matches, while a string with two different numbers of `o`s does not.
```

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5f20cc9584cada942a4.md:
```diff
@@ -89,19 +108,25 @@ To define a named capture group, you add a question mark (`?`) followed by the n
 const regex = /free(?<code>co+de)camp.*free\1camp/i;
```
Redundant quotes around `code` in the sentence above.

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5f20cc9584cada942a4.md:
````diff
@@ -89,19 +108,25 @@ To define a named capture group, you add a question mark (`?`) followed by the n
 ```
 
-Now we can update our backreference in the regular expression to refer to this group. A named back reference starts with a backslash followed by the letter `k` in JavaScript. Then you add the name, again enclosed in less than and greater than signs. Let's take a look at that:
+Now we can update our backreference in the regular expression to refer to this group. A named back reference starts with a backslash followed by the letter `k` in JavaScript. Then you add the name, again enclosed in less than (`<`) and greater than (`>`) signs. Let's take a look at that:
````
```suggestion
Now we can update our backreference in the regular expression to refer to this group. A named backreference starts with a backslash followed by the letter `k` in JavaScript. Then you add the name, again enclosed in less than (`<`) and greater than (`>`) signs. Let's take a look at that:
```

Dario-DC commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5f20cc9584cada942a4.md:
````diff
@@ -89,19 +108,25 @@ To define a named capture group, you add a question mark (`?`) followed by the n
+```
 
 Finally, sometimes you want to create a group of characters, but don't need the captured result.
 
````
It seems `You can't backreference a non-capturing group.` towards the end of the transcript is not part of the video. I cannot make the suggestion in the right place.

Dario-DC reviewed (commented):
Left a few suggestions, mostly to combine short sentences into single paragraphs since we have done that for the other transcripts.

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5ba834ded4bb067e67c.md:
```diff
@@ -10,6 +10,243 @@ dashedName: what-are-some-common-regular-expression-modifiers-used-for-searching
 
 Watch the video lecture and answer the questions below.
```
```suggestion
Watch the video or read the transcript and answer the questions below.
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5c549775c4be710237c.md:
```diff
@@ -10,6 +10,222 @@ dashedName: how-can-you-match-and-replace-all-occurrences-in-a-string
 
 Watch the video lecture and answer the questions below.
```
```suggestion
Watch the video or read the transcript and answer the questions below.
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5d0048bb74c18431296.md:
```diff
@@ -1,32 +1,36 @@
 # --description--
 
-The video for this lecture isn't available yet, one will be available soon. Here is a transcript of the lecture for now:
+Watch the video lecture and answer the questions below.
```
```suggestion
Watch the video or read the transcript and answer the questions below.
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5dc74176e4c496d09e6.md:
```diff
@@ -1,14 +1,18 @@
 # --description--
 
-The video for this lecture isn't available yet, one will be available soon. Here is a transcript of the lecture for now:
+Watch the video lecture and answer the questions below.
```
```suggestion
Watch the video or read the transcript and answer the questions below.
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5e54e3a154c8078ed48.md:
```diff
@@ -1,18 +1,22 @@
 # --description--
 
-The video for this lecture isn't available yet, one will be available soon. Here is a transcript of the lecture for now:
+Watch the video lecture and answer the questions below.
```
```suggestion
Watch the video or read the transcript and answer the questions below.
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733c5f20cc9584cada942a4.md:
```diff
@@ -1,50 +1,62 @@
 # --description--
 
-The video for this lecture isn't available yet, one will be available soon. Here is a transcript of the lecture for now:
+Watch the video lecture and answer the questions below.
```
```suggestion
Watch the video or read the transcript and answer the questions below.
```

jdwilkin4 commented on curriculum/challenges/english/25-front-end-development/lecture-working-with-regular-expressions/6733aad43b3ebff588a26fb5.md:
````diff
@@ -10,6 +10,172 @@ dashedName: what-are-regular-expressions-and-what-are-some-common-methods
+const regex = /freeCodeCamp/;
+```
+
+Notice how, in JavaScript, you define a regular expression by creating your pattern between two backslashes (`/`). Try not to confuse this with a comment, where the text comes after both backslashes.
````
https://github.com/freeCodeCamp/freeCodeCamp/issues/59047

jdwilkin4 committed:
Apply suggestions from code review

Co-authored-by: Dario-DC <105294544+Dario-DC@users.noreply.github.com>

jdwilkin4 committed:
fix: review suggestions
